### PR TITLE
api: returning json, accepting parameter for number of questions

### PR DESCRIPTION
**Pull request by sousajo-cc**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,21 +1,99 @@
...
+    static ref QUESTION_KV_DATABASE: HashMap<u32, QuestionAnswer> = {
         let mut m = HashMap::new();
-        m.insert(0, ("What's my name?", "Acer\n"));
-        m.insert(1, ("What's Luiz0s name?", "Luiz\n"));
-        m.insert(2, ("What's Jorge's name?", "GOD\n"));
-        m.insert(3, ("Are you sentient?", "On Ma I toN\n"));
-        m.insert(4, ("Where do you live?", "Stalker!\n"));
-        m.insert(5, ("What's the purpose of life?", "FU\n"));
-        m.insert(6, ("How to make a good question?", "That's not the way!\n"));
+        m.insert(
```
Comment (luizchagasjardim) [inline src/main.rs]: Maybe use HashMap::from() instead of inserting.

File: src/main.rs [hunk 2]
```diff
@@ -1,21 +1,99 @@
...
+            0,
+            QuestionAnswer {
+                question: "What's my name?".to_string(),
+                answer: "Acer".to_string(),
+            },
+        );
+        m.insert(
+            1,
+            QuestionAnswer {
+                question: "What's Luiz0s name?".to_string(),
```
Comment (luizchagasjardim) [inline src/main.rs]:
```suggestion
                question: "What's Luiz's name?".to_string(),
```

File: src/main.rs [hunk 3]
```diff
@@ -1,21 +1,99 @@
...
+            QuestionAnswer {
+                question: "What's Jorge's name?".to_string(),
+                answer: "GOD".to_string(),
+            },
+        );
+        m.insert(
+            3,
+            QuestionAnswer {
+                question: "Are you sentient?".to_string(),
+                answer: "On Ma I toN".to_string(),
```
Comment (luizchagasjardim) [inline src/main.rs]:
```suggestion
                answer: "Ton Ma I oN".to_string(),
```

File: src/main.rs [hunk 4]
```diff
@@ -25,36 +103,39 @@ async fn hello_acer() -> impl Responder {
...
 
-#[get("/get_random_answers")]
-async fn get_random_answers() -> impl Responder {
+#[get("/random_questions/{number_of_questions}")]
+async fn random_questions(
+    number_of_questions: web::Path<usize>,
+) -> Result<impl Responder, AcerError> {
     let mut rng = thread_rng();
     let questions_database_size = QUESTION_KV_DATABASE.keys().len();
+    let number_of_questions_ = *number_of_questions;
```
Comment (luizchagasjardim) [inline src/main.rs]:
```suggestion
    let number_of_questions = *number_of_questions;
```
Rust has no problem with redefinition. The last one is the one that counts.

**Review by luizchagasjardim — APPROVED** [review]
Very good work!

File: src/main.rs (hunk 1 above)
Comment (sousajo-cc) [reply src/main.rs]: great ty !!!

File: src/main.rs (hunk 4 above)
Comment (sousajo-cc) [reply src/main.rs]: great, thankssssssssssssssss

**Pull request merged**